### ISSUE TITLE
[BUG] Error running command dodona.exercise.create

### ISSUE THREAD

**Issue by Stijn0303** [thepieterdc/dodona-plugin-vscode#481]
# Description
Error running command dodona.exercise.create: Cannot read properties of undefined (reading 'replace'). This is likely caused by the extension that contributes dodona.exercise.create.

# Environment
- OS: [Windows] (Edition: Windows 10 Home)
- Version: 21H2
- Include a screenshot of the `Help -> About` screen.
<img width="960" alt="2022-10-01 help - about" src="https://user-images.githubusercontent.com/114802446/193401526-53df1a8a-3b7d-4344-af50-b771242be1ab.png">

# Screenshots
<img width="960" alt="2022-10-01 Bug" src="https://user-images.githubusercontent.com/114802446/193401559-421126b3-ad32-42cd-b960-be34f98f6076.png">

**Comment by chvp** [comment]
You probably want to delete the API token you posted in your screenshot.

**Comment by stijndcl** [comment]
Your second screenshot leaks your Dodona token, you should go to the Dodona panel and delete it ASAP.


Did you open a folder in VSCode to work in, or do you not have an active workspace?

An update was released yesterday that should show a pop-up that hints at this. Do you have the latest version of the extension?

**Comment by Stijn0303** [comment]
My mistake... To change the token, you just need to remove the old one and make a new one right?
I have never worked with VSCode before, so I don't really know if I have an active workspace or not.
Where can I check if I have the latest version?

**Comment by stijndcl** [comment]
> My mistake... To change the token, you just need to remove the old one and make a new one right?

Yes, click the trash can button on the right hand side to delete it. Then generate a new one and paste it in your plugin settings.

![image](https://user-images.githubusercontent.com/60451863/193402503-91471b0d-9913-4e8f-b916-316b6daacf33.png)

> I have never worked with VSCode before, so I don't really know if I have an active workspace or not. 

Click the first button in your sidebar (Explorer):
![image](https://user-images.githubusercontent.com/60451863/193402539-a9a038c1-de77-46ea-b967-83e29d93e152.png)

Do you see a message that says "You have not yet opened a folder. [OPEN FOLDER]" - or do you see files?
![image](https://user-images.githubusercontent.com/60451863/193402560-f340215a-10ff-4df0-ab36-5a881d847ade.png)

> Where can I check if I have the latest version?

Click the Extensions button in your sidebar:
![image](https://user-images.githubusercontent.com/60451863/193402571-53262168-bbaf-4032-9ea8-a2e63af048e8.png)

The Dodona extension should not have an `Update` button visible.

At the top of the menu, click "Check for extension updates":
![image](https://user-images.githubusercontent.com/60451863/193402666-1901991e-ef03-46c9-b69e-8058efda9e38.png)

**Comment by Stijn0303** [comment]
I changed my API code.

I only see files I think. 

My extensions are up to date.

**Comment by stijndcl** [comment]
Can you send the traceback of the error so we can see where it came from?

- Open the `OUTPUT` tab at the bottom of your screen
- In the dropdown, select `Log (Extension Host)`

![image](https://user-images.githubusercontent.com/60451863/193403309-99d3d540-07b3-4df0-8007-e757578d1771.png)

When you now create an exercise, you should see a big error message in there. Could you send that over?

It should look something like

```
[2022-10-01 11:42:11.389] [exthost] [error] TypeError: Cannot read properties of undefined (reading 'replace')
    at ...
    at ...
    at ...
    at ...
    at ...
    at ...
```

**Comment by Stijn0303** [comment]
What do you mean by creating an exercise, because when I just try to open one, I get the error, but nothing seems to appear in the Log (Extension Host)...

**Comment by stijndcl** [comment]
I meant clicking on an exercise in the Dodona tab to open it. It's very strange that you don't see anything in the logs. I can't reproduce this issue myself, so without a traceback it's a bit difficult to pinpoint the cause of this.

I'll investigate a bit more.
